PennMobile/Home/Controllers/DiningCellSettingsController.swift

```diff
@@ -73,43 +56,26 @@ class DiningCellSettingsController: UITableViewController {
     }
 
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        let cafe = cafes[indexPath.row]
-
-        if !chosenCafes.contains(cafe) {
-            if let cell = tableView.cellForRow(at: indexPath) {
-                cell.accessoryType = .checkmark
-            }
-            chosenCafes.append(cafe)
-        }
-    }
-
-    override func tableView(_ tableView: UITableView, didDeselectRowAt indexPath: IndexPath) {
-        let cafe = cafes[indexPath.row]
-
-        if chosenCafes.contains(cafe) {
-            if let cell = tableView.cellForRow(at: indexPath) {
-                cell.accessoryType = .none
-            }
-
-            if let index = chosenCafes.firstIndex(of: cafe) {
-                chosenCafes.remove(at: index)
-            }
+        if chosenVenueIds.contains(allVenues[indexPath.row].id) {
+            chosenVenueIds.remove(allVenues[indexPath.row].id)
+        } else {
+            chosenVenueIds.insert(allVenues[indexPath.row].id)
         }
+        tableView.reloadRows(at: [indexPath], with: .automatic)
     }
 
     @objc func handleCancel() {
         self.dismiss(animated: true, completion: nil)
-        FirebaseAnalyticsManager.shared.trackEvent(action: .updateDiningPreferences, result: .cancelled, content: chosenCafes.count)
+        FirebaseAnalyticsManager.shared.trackEvent(action: .updateDiningPreferences, result: .cancelled, content: chosenVenueIds.count)
```

Comment:
Let's remove the Firebase logs since they aren't relevant anymore. We track this information in the DB

Reply:
Done